src/Services/SettingsBase.cs

```diff
@@ -229,10 +249,10 @@ private static async Task<string> ReadFileAsStringAsync(IFileData file, Cancella
 
             // Convert to raw bytes
             var typeFileBytes = new byte[typeFileStream.Length];
-            await typeFileStream.ReadAsync(typeFileBytes, 0, typeFileBytes.Length, token);
+            var read = await typeFileStream.ReadAsync(typeFileBytes, 0, typeFileBytes.Length, token);
 
             // Read bytes as string
-            return Encoding.UTF8.GetString(typeFileBytes);
+            return Encoding.UTF8.GetString(typeFileBytes.Take(read).ToArray());
```

Comment:
This is an extra operation that doesn't have any functional effect. The byte array is initialized at the exact needed size above, and the entire array is filled using data from the stream - where we've explicitly asked it to return enough bytes to fill our buffer.

Reply:
![image](https://user-images.githubusercontent.com/53011783/184421819-ec6a28e6-5a13-4efb-9d15-e85f9b5c36f3.png)

Reply:
I see, that makes sense. In that case, rather than invoking linq every time, we should add a `Guard` to make sure the number of actually read bytes are what we expect -- have it throw otherwise, which will be caught and emitted on the `SettingLoadFailed` event.